openstack/barbican/templates/barbican-nanny-deployment.yaml

```diff
@@ -0,0 +1,112 @@
+    rollingUpdate:
+      maxUnavailable: {{ .Values.pod.lifecycle.upgrades.deployments.rollingUpdate.maxUnavailable }}
+      maxSurge: {{ .Values.pod.lifecycle.upgrades.deployments.rollingUpdate.maxSurge }}
+    {{ end }}
+  selector:
+    matchLabels:
+      app: barbican-nanny
+  template:
+    metadata:
+      labels:
+        app: barbican-nanny
+        name: barbican
+        alert-tier: os
+        alert-service: barbican
+        {{- include "utils.topology.pod_label" . | indent 8 }}
+      annotations:
+        chart-version: {{.Chart.Version}}
+        checksum/etc-configmap.conf: {{ include (print $.Template.BasePath "/etc-configmap.yaml") . | sha256sum }}
+        secrets-hash: {{ include (print $.Template.BasePath "/secrets.yaml") . | sha256sum }}
+        {{- if .Values.proxysql.mode }}
+        prometheus.io/scrape: "true"
+        prometheus.io/targets: {{ required ".Values.alerts.prometheus missing" .Values.alerts.prometheus | quote }}
+        {{- end }}
+        {{- include "utils.linkerd.pod_and_service_annotation" . | indent 8 }}
+    spec:
+      {{ include "utils.proxysql.pod_settings" . | indent 6 }}
+{{- if .Values.barbican_nanny.db_secret_move.enabled }}
+      containers:
+      - name: db-check
+        image: {{ required ".Values.global.registry is missing" .Values.global.registry }}/loci-barbican:{{.Values.imageVersionBarbicanApi | default .Values.imageVersionBarbicanApi | required "Please set barbican.imageVersion or similar"}}
+        imagePullPolicy: IfNotPresent
+        command:
+        - dumb-init
+        - kubernetes-entrypoint
+        env:
+        - name: COMMAND
+          value: "{{ if not .Values.barbican_nanny.debug }}/bin/bash /scripts/barbican-db-check.sh{{ else }}sleep inf{{ end }}"
+        - name: BARBICAN_DB_SECRET_MOVE_ENABLED
+          value: {{ .Values.barbican_nanny.db_secret_move.enabled | quote }}
+        - name: BARBICAN_DB_OLD_PROJECT_ID
+          value: {{ .Values.barbican_nanny.db_secret_move.old_project_id | quote }}
+        - name: BARBICAN_DB_NEW_PROJECT_ID
+          value: {{ .Values.barbican_nanny.db_secret_move.new_project_id | quote }}
+        - name: BARBICAN_NANNY_INTERVAL
+          value: {{ .Values.barbican_nanny.interval | quote }}
+        {{- if .Values.sentry.enabled }}
+        - name: SENTRY_DSN
+          valueFrom:
+            secretKeyRef:
+              name: sentry
+              key: barbican.DSN.python
+        {{- end }}
+        resources:
+          requests:
+            memory: "250Mi"
+            cpu: "25m"
+          limits:
+            memory: "250Mi"
+            cpu: "100m"
+        volumeMounts:
+        - name: barbican-scripts
+          mountPath: /scripts/barbican-db-check.sh
+          subPath: barbican-db-check.sh
+          readOnly: true
+        - name: barbican-etc
+          mountPath: /etc/barbican/barbican-api.conf
+          subPath: barbican-api.conf
+          readOnly: true
```

Comment:
Running `barbican-manage`, I don't think it will find `barbican-api.conf`. Afaik, `oslo.config` will use the name of the command (i.e. `barbican-manage.conf`) and the name of the project (i.e. `barbican.conf`) automatically.

Reply:
could you please elaborate ? it works well in qa-de-3.

Reply:
If it runs well, maybe it doesn't need any setting from the `barbican-api.conf`. Did you check that any of these settings take effect?
To my knowledge, `barbican-api.conf` will not be consumed by an executable called `barbican-manage` unless explicitly given via `--config-file` option.